eclipse-collections-code-generator/src/main/resources/api/primitiveIterable.stg

```diff
@@ -79,16 +80,52 @@ public interface <name>Iterable extends PrimitiveIterable
     boolean contains(<type> value);
 
     /**
-     * Returns true if the all of the values specified in the source array are contained
+     * Returns true if all of the values specified in the source array are contained
      * in the <name>Iterable, and false if they are not.
      */
-    boolean containsAll(<type>... source);
+    default boolean containsAll(<type>... source)
+    {
+        if (this.size() \<= 32 || source.length \< 4)
```

Comment:
This could be simplified, with a small cost.

```
default boolean containsAll(<type>... source)
{
    return this.containsAll(new <name>ArrayList(source));
}

```

Reply:
That's not a good tradeoff for small arrays. I'd rather keep the code as is.

Reply:
This is one of those places where having primitive array utility classes with some basic patterns would have helped. Something for another day perhaps. Please feel free to merge.